packages/app/src/cli/utilities/developer-platform-client/app-management-client.ts

```diff
@@ -217,7 +219,9 @@ export class AppManagementClient implements DeveloperPlatformClient {
     return (await this.session()).accountInfo
   }
 
-  async appFromId(appIdentifiers: MinimalAppIdentifiers): Promise<OrganizationApp | undefined> {
+  async appFromIdentifiers(
+    appIdentifiers: MinimalAppIdentifiersPossiblyExcludingId,
+  ): Promise<OrganizationApp | undefined> {
```

Comment:
Can't we change `appFromIdentifiers` to need just `api-key`? (and org-id temporarily while it's removed)
There is no need for this method to have the `id` optional when most use cases will only pass api-key.

There a couple cases where we do have the `id` available, but if it's not needed, why not just simplify everything?